### ISSUE TITLE
Receiving side shouldn't mount datasets

### ISSUE THREAD

**Issue by zviratko** [aaronhurt/zfs-replicate#48]
By default, if a dataset has canmount=on and mountpoint set, it will get mounted, this is highly undesirable in a backup pool.
I usually worked around thet when making backups by setting canmount=noauto/off on the parent dataset I receive the backup into, but this doesn't work when the source dataset has this set explicitely (not inherited), like FreeBSD zroot datasets. Ouch.

At the very least, zfs-replicate should by default use "zfs recv -u" so that the datasets are not mounted and one can change their properties, even better it should support "zfs receive -o canmount=noauto" (or maybe -x?) from a config file so that it can be safely received by absent minded admin who will forget to change those after the receive is done the first time (which can take some time).

Not sure if there's a better way to do this.

Thanks.

**Comment by aaronhurt** [comment]
I don't think it needs a new variable, modifying the existing pipes should be sufficient.

**Comment by aaronhurt** [comment]
Closing due to inactivity, @zviratko please feel free to re-open if the suggested path doesn't resolve your issue.

**Comment by zviratko** [comment]
I apologise for not following up.

Yes, looks like those existing variables are enough, thank you.

I'd still like to see a sane default or an explicit setting for this, though :)
